Added script to offload drone videos and rename to epoch time

PhillipShahviri opened the pull request.

PeterMalkin reviewed (changes requested):
1. Why use "video_data" vs "data"? 
2. I do not see the device id mentioned anywhere. As it is, the video data from all sources will be mixed together.
3. The videos are identified by .mp4 extensions. What if the video format is different? mks? mov? avi?

PhillipShahviri commented:
@PeterMalkin 

1. The sensor data offloading script uses "data", so I didn't want to conflict with that name. Do you think it would be better to put all the videos in subfolder(s) within the "data" folder?

2. Since the video data is not specific to any one tag, or even any one drone (one storage device could be used to offload videos from multiple drones), I wasn't sure how to separate the files. What if we just separated them based on the date the videos were captured?

3. Thanks for catching this. My recent commit addresses it.

PeterMalkin commented:
1. Correct. Lets keep all sources of data inside /data, since there are assumptions made inside the s3upload scripts that use /data as the source of data. The distinction between the type of data can be made by the extension of the files.
2. I see the benefit to keep around the data about which data was captured by which devices for several reasons. Imagine, one drone manufacturer has a bug and all the videos coming from that drone are somehow corrupt, or whave the metadata wrong, or sometheinge else. It feels that loosing this data is disadvantageous. FOr all other devices that we use for data collection we always attribute the data that came from the device to a unique id of the device that did the data collection.

PhillipShahviri commented:
With the new commit, videos will be offloaded and placed in data/<device_id> .

device_id is given by the user and then checked to make sure it exists already in "Device ID List.txt" which is in the raw folder on s3. If it is not already in that list then the offload is rejected with an appropriate error message.  

I tested that the videos upload nicely in their appropriate folder when using the existing s3upload command

The raw->flac converter script was also edited. Currently it just creates a flac folder in the previous directory and dumps the converted files in there. I changed it so that it creates folders within the flac directory that separate the data based on the capturing device. This way one can just point to the flac directory when running s3upload. I also adjusted the naming so we no longer get .raw.flac files

PeterMalkin commented on README.md:
````diff
@@ -171,10 +196,11 @@ BUMP_LEVEL=minor make release
 ```console
 #!/bin/bash
-OUTDIR="../flac"
+OUTDIR="../flac/${PWD##*/}"
````
This will create both files in the same folder. If subsequently you use `ceti s3upload {path}`, it will upload both raw files and .flac files. Not sure what's the best way to deal with this. The easiest is of course to `rm $f` after encoding. I guess its not too big of a deal, since this is just inside readme?

PhillipShahviri replied:
Not sure I completely follow. There are no .raw files in the 'flac' folder that this script creates. If someone runs `ceti s3upload data/flac` only the .flac files will be uploaded.

PeterMalkin commented on README.md:
```diff
@@ -171,10 +196,11 @@ BUMP_LEVEL=minor make release
 mkdir -p "$OUTDIR"
 for f in *.raw; do
-  flac --channels=3 --bps=16 --sample-rate=96000 --sign=signed --endian=big --force-raw-format "$f" --force --output-name="$OUTDIR/$f.flac" &
+  IFS='.' read -r -a split_filename <<< "$f"
```
I recommend $(realpath "$PWD") instead of '.'

PhillipShahviri replied:
Tried this and it is not quite giving me the results I want.

Previously this script was making .raw.flac files. This is part of the solution to eliminate this. I am just trying to separate the timestamped filename into 2 strings based on the dot delimiter '.'
This way I can just reference the first string in the returned list when I do ```--output-name=$OUTDIR/${split_filename[0]}.flac"```

PeterMalkin reviewed (commented):
video_offload is LGTM, but I am curious whether the naming is ok.
In a sense you are creating a fairly generic way to upload arbitrary files to S3 as long as the device ID is added in the allow list. 
So I wonder if we should use a more generic name (say "stage_aux_data", or some such) or just stick with video_offload?

PhillipShahviri commented:
I see what you're saying about the naming. I think it's a good idea. I will refactor and generalize for more generic data.
